network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java

```diff
@@ -0,0 +1,107 @@
+package com.capacitorjs.plugins.network;
+
+import android.Manifest;
+import android.net.NetworkInfo;
+import com.getcapacitor.JSObject;
+import com.getcapacitor.Logger;
+import com.getcapacitor.NativePlugin;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginCall;
+import com.getcapacitor.PluginMethod;
+
+@NativePlugin(name = "Network", permissions = { Manifest.permission.ACCESS_NETWORK_STATE })
+public class NetworkPlugin extends Plugin {
+    private Network implementation;
+    public static final String NETWORK_CHANGE_EVENT = "networkStatusChange";
+    private static final String PERMISSION_NOT_SET = Manifest.permission.ACCESS_NETWORK_STATE + " not set in AndroidManifest.xml";
+
+    /**
+     * Monitor for network status changes and fire our event.
+     */
+    @Override
+    @SuppressWarnings("MissingPermission")
+    public void load() {
+        implementation = new Network(getContext());
+        implementation.setStatusChangeListener(this::updateNetworkStatus);
+    }
+
+    /**
+     * Clean up callback to prevent leaks.
+     */
+    @Override
+    protected void handleOnDestroy() {
+        implementation.setStatusChangeListener(null);
+    }
+
+    /**
+     * Get current network status information.
+     * @param call
+     */
+    @SuppressWarnings("MissingPermission")
+    @PluginMethod
+    public void getStatus(PluginCall call) {
+        if (hasRequiredPermissions()) {
+            call.resolve(getStatusJSObject(implementation.getNetworkStatus()));
+        } else {
+            call.reject(PERMISSION_NOT_SET);
+        }
```

Comment:
I may not be needed under normal conditions but it shouldn't cause any harm to leave it either. And it could provide more useful errors in the unlikely case that the build configuration really becomes messed up.